internal/rbd/types/volume.go

```diff
@@ -40,4 +42,23 @@ type Volume interface {
 
 	// RemoveFromGroup removes the Volume from the VolumeGroup.
 	RemoveFromGroup(ctx context.Context, vg VolumeGroup) error
+
+	// GetPoolName returns the name of the pool where the volume is stored.
+	GetPoolName() string
+	// GetCreationTime returns the creation time of the volume.
+	GetCreationTime() (*timestamppb.Timestamp, error)
+	// GetMetadata returns the value of the metadata key from the volume.
+	GetMetadata(key string) (string, error)
+	// SetMetadata sets the value of the metadata key on the volume.
+	SetMetadata(key, value string) error
+	// RepairResyncedImageID updates the existing image ID with new one in OMAP.
+	RepairResyncedImageID(ctx context.Context, ready bool) error
+	// HandleParentImageExistence checks the image's parent.
+	// if the parent image does not exist and is not in trash, it returns nil.
+	// if the flattenMode is FlattenModeForce, it flattens the image itself.
+	// if the parent image is in trash, it returns an error.
+	// if the parent image exists and is not enabled for mirroring, it returns an error.
+	HandleParentImageExistence(ctx context.Context, flattenMode FlattenMode) error
+
+	Mirror
```

Comment:
Comment on this please, and move it to the top so that it is immediately clear that everything from `Mirror` is inherited too.

Or, maybe introduce a `MirroredVolume` type that inherits both from `Volume` and `Mirror`? That way the `Volume` interface stays small and simple for 'normal' usage, the `rbdVolume` object can still implement both.

Reply:
Done. Moved it to top.